cmd/up/exec_test.go

```diff
@@ -494,6 +516,93 @@ func TestGetEnvsFromConfigMap(t *testing.T) {
 	}
 }
 
+func TestGetEnvsFromDevContainer(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		name         string
+		expectedEnvs []string
+		podspec      *apiv1.PodSpec
+		client       kubernetes.Interface
+	}{
+		{
+			name: "dev container without env vars",
+			podspec: &apiv1.PodSpec{
+				Containers: []apiv1.Container{
+					{
+						Env: []apiv1.EnvVar{},
+					},
+				},
+			},
+		},
+		{
+			name: "dev container with regular env var",
+			podspec: &apiv1.PodSpec{
+				Containers: []apiv1.Container{
+					{
+						Env: []apiv1.EnvVar{
+							{
+								Name:  "FROMPOD",
+								Value: "VALUE1",
+							},
+						},
+					},
+				},
+			},
+			expectedEnvs: []string{
+				"FROMPOD=VALUE1",
+			},
+		},
+		{
```

Comment:
Can we also test what happens when the secret doesn't exist? 

Also, testing this scenario I noticed that it could cause a very weird error, that doesn't provide enough information to the user to be able to troubleshoot it by himself:

<img width="850" alt="Captura de pantalla 2023-08-23 a las 16 56 20" src="https://github.com/okteto/okteto/assets/25170843/c6ae8c73-5e36-485c-92b1-f591efcdae29">

Reply:
Nice finding @jLopezbarb - I've wrapped the error with some more explaination for the user. Do you have any further ideas to improve this?
![Screenshot 2023-08-23 at 19 07 51](https://github.com/okteto/okteto/assets/2318450/3bb37988-57fe-42fc-93dd-b70832fb1d4a)

Reply:
> Can we also test what happens when the secret doesn't exist?

💯 I've added a test and also modified the return value to return `envs` calculated up to the moment of the error. In the test added, it showcases that the first env var is returned, then it fails because of the secret, and the last env var is not available either despite it would have been available.

I am wondering if we like this logic, or we rather log the error but continue. I guess it can be revisited in a future PR because we would need to revisit all the other functions that return err too.

Commit to check: https://github.com/okteto/okteto/pull/3931/commits/ec854fc9a72c0031eeae274cd37018930bec1062

Reply:
Following with what k8s does. It creates the pod but then fails on initialization. I think it's better to fail fast and could be revisited later.

> Do you have any further ideas to improve this?

I think the error is very explicit and very useful, maybe adding a `UserError` with a `Hint` let us shorten the first part but I think is good as it is